client/my-sites/site-settings/header-code/index.jsx

```diff
@@ -0,0 +1,133 @@
+import {
+	isRequestingSiteSettings,
+	isSavingSiteSettings,
+	isSiteSettingsSaveSuccessful,
+	getSiteSettingsSaveError,
+} from 'calypso/state/site-settings/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import './style.scss';
+
+const HeaderCodeSettings = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ editedCode, setEditedCode ] = useState( '' );
+
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const headCode = useSelector( ( state ) =>
+		getSiteSetting( state, siteId, 'jetpack_header_code' )
+	);
+	const isAdmin = useSelector( ( state ) => canCurrentUser( state, siteId, 'manage_options' ) );
+	const isRequestingSettings = useSelector( ( state ) =>
+		isRequestingSiteSettings( state, siteId )
+	);
+	const isSaveFailure = useSelector( ( state ) => getSiteSettingsSaveError( state, siteId ) );
+	const isSaveSuccess = useSelector( ( state ) => isSiteSettingsSaveSuccessful( state, siteId ) );
+	const isSavingSettings = useSelector( ( state ) => isSavingSiteSettings( state, siteId ) );
+	const siteSlug = useSelector( ( state ) => getSelectedSiteSlug( state, siteId ) );
+	const saveSettings = () => {
+		dispatch(
+			saveSiteSettings( siteId, {
+				jetpack_header_code: editedCode,
+			} )
+		);
+	};
+
+	useEffect( () => {
+		const noticeOptions = {
+			duration: 4000,
+		};
+
+		if ( isSaveSuccess ) {
+			dispatch( successNotice( translate( 'Code updated successfully!' ), noticeOptions ) );
+		} else if ( isSaveFailure ) {
+			dispatch( errorNotice( translate( 'There was an error saving your code.' ), noticeOptions ) );
+		}
+	}, [ isSaveSuccess, isSaveFailure ] );
+
+	return (
+		<Main className="header-code">
+			<DocumentHead title={ translate( 'Header Code' ) } />
+			{ siteId && <QuerySiteSettings siteId={ siteId } /> }
+			{ ! isAdmin && (
+				<EmptyContent
+					illustration={ illustration404 }
+					title={ translate( 'You are not authorized to view this page' ) }
+				/>
+			) }
+
+			{ isAdmin && (
+				<>
+					<NavigationHeader
+						title={ translate( 'Header Code' ) }
+						subtitle={ translate( 'Insert HTML meta tags and JavaScript code to your site.' ) }
+					/>
+					<HeaderCake backHref={ `/settings/general/${ siteSlug }` } isCompact>
+						<h1>{ translate( 'Header Code' ) }</h1>
+					</HeaderCake>
+					<Card>
+						<p className="header-code__text">
+							{ translate(
+								'The following code will be inserted within {{code}}<head>{{/code}} on your site. {{link}}Learn more{{/link}}.',
+								{
+									components: {
+										code: <code />,
+										link: (
+											<InlineSupportLink supportContext="insert-header-code" showIcon={ false } />
+										),
+									},
+								}
+							) }
+						</p>
+
+						<FormTextarea
+							className={ classNames( 'header-code__editor', {
+								'is-loading': isRequestingSettings,
+							} ) }
+							defaultValue={ headCode }
```

Comment:
Is there a good reason why we're using `defaultValue` and not `value`? If we're not using controlled components, it may be unnecessary to store the edited code in the component state. Similarly, if we need to use a component state, then we likely want to use a controlled component. Furthermore, is there a chance the value could get out of sync since the user is changing something here, but a new code was already saved and returned by the API?

Reply:
Switched to `value` which also allows disabling the button when it's unedited. I don't _think_ that this approach will allow the value to get out of sync.